box: fix unique violation in functional index with nullable parts (#8672)
tarantool/tarantool: changelogs/unreleased/gh-8587-unique-violation-in-functional-index-with-nullable.md

## bugfix/core

> * Fixed a bug when a tuple can be inserted even though it violates the `unique`
>   constraint of a functional index, which has a nullable part (gh-8587).

Reviewer: ```suggestion
* Fixed a bug when a tuple could be inserted even if it violates a `unique`
  constraint of a functional index that has a nullable part (gh-8587).
```

Author: Done.
